### PR TITLE
Feature: real-time video playing within provided attachment

### PR DESCRIPTION
**Pull request by sskorol**
* added 3 new attachment types: `mp4`, `webm`, `ogg`, which supported by html5 video (affected attachmentType.js and AttachmentView.hbs);
 * added dummy data for corresponding formats;
 * updated a test which checks attachments amount;
 * fixed allure-report-face tests running issue on Windows (`jasmine` command should be first).

See initial request [#624](https://github.com/allure-framework/allure-core/issues/624)

### CODE REVIEW

**Comment by sskorol** [comment]
Seems like `jasmine` fix, which works for Windows, doesn't work on Ubuntu 15.10 (btw, reverted change doesn't work as well). Smth wrong with cross-platform `JASMINE_CONFIG_PATH` resolving.  Will double check it. Please don't merge it yet.

File: allure-report-face/src/components/attachment/AttachmentView.hbs [hunk 1]
```diff
@@ -38,6 +38,27 @@
     <div class="pane__section">
         <object class="attachment__embed" type="image/svg+xml" data="{{sourceUrl}}">Your browser does not support SVG</object>
     </div>
+{{else if (eq type "mp4")}}
+    <div class="pane__section">
```
Comment (just-boris) [inline allure-report-face/src/components/attachment/AttachmentView.hbs]: why not just pass mime-type into the template to the `source` tag.


```hbs
{{else if (eq type "video")}}
    <div class="pane__section">
         <video controls>
             <source src="{{sourceUrl}}" type="{{type}}">
             Your browser does not support video tag
         </video>
     </div>
{{/if}}
```

One `else if` statement better than many

**Comment by just-boris** [comment]
Fixed test command using [`cross-env`](https://github.com/kentcdodds/cross-env) very useful thing for these cases.

**Comment by just-boris** [comment]
1) Your test video attachments are so big (~1Mb). I don't think we should test browsers abilities about video formats support. One small video is enough as an example.

2) I personally don't like pull requests from fork to another. It makes easier to lose this changes. Your feature is good, so please hang on and then we can merge it right to the master in core repo.

According to https://github.com/allure-framework/allure-core/issues/721#issuecomment-206445080, merge this is the main goal for @baev now. It can't be done with magical "Merge button", master branch has been changed and needs to resolve conflicts manually. But still, I think this is the only way to go.

**Comment by sskorol** [comment]
@just-boris will check this fix now, thanks. 

Btw, also faced with report rendering on Ubuntu 15.10. It gives me `Cannot read property 'radio' of undefined` on marionette (both during installation and in browser). Seems like a popular bug. However, I couldn't resolve it yet (tried to change dependencies versions, no luck). Any idea on this?

Regarding PR itself, @baev gave me a link on your repo (in gitter) and asked to create PR against new-face. That's why I did this. However, I'm fine to wait until it's merged to master and re-create PR.

**Comment by just-boris** [comment]
> Btw, also faced with report rendering on Ubuntu 15.10. It gives me Cannot read property 'radio' of undefined
Yes, I see that bug and  working on it now

**Comment by just-boris** [comment]
Fixed. During this time, some dependencies were outdated and my attention was needed to update it.

**Comment by sskorol** [comment]
@just-boris pulled latest updates, still getting those [error](https://drive.google.com/open?id=0B1ORH0iP6-geWXVPTkNiSG5FOEE) in browser and while [installation](https://drive.google.com/open?id=0B1ORH0iP6-geZ1Q5S0Z1Z2ctVmc).

**Comment by just-boris** [comment]
`rm -rf node_modules` folder should solve that issue

File: allure-report-face/src/components/attachment/AttachmentView.hbs (hunk 1 above)
Comment (sskorol) [inline allure-report-face/src/components/attachment/AttachmentView.hbs]: A bit tricky case. With this suggestion we'll be moved to final `else` block, as there's no such type = `video` in `attachmentType.js`. 

If we force `attachmentType.js` to return just `video` to reach this `else if` block, as a result our source type will be invalid and video won't be rendered.

To support your suggestion I can force `attachmentType.js` to return just original `type` foreach new video format. Also it'd be required to add new js helper to support `contains` operation, as pure `eq` won't work. If you're ok with it, I'll push this update shortly.

**Comment by sskorol** [comment]
@just-boris great, above command resolved the issue. Looking at your comments now.

File: allure-report-face/src/util/attachmentType.js [hunk 2]
```diff
@@ -26,6 +26,12 @@ export default function typeByMime(type) {
             return 'csv';
         case 'image/svg+xml':
             return 'svg';
+        case 'video/mp4':
+            return 'mp4';
+        case 'video/ogg':
+            return 'ogg';
+        case 'video/webm':
+            return 'webm';
```
Comment (just-boris) [inline allure-report-face/src/util/attachmentType.js]: I'd like to write this:

```
case 'video/mp4':
case 'video/ogg':
case 'video/webm':
    return 'video';
```

You can always get actual attachment MIME-type in the template via `{{attachement.type}}` to pass it to `source` tag

Comment (sskorol) [inline allure-report-face/src/util/attachmentType.js]: Aha, didn't know about `{{attachement.type}}` and `{{type}}` difference to be honest. I'm new to Allure yet. Thanks, seems fine to me.

**Comment by sskorol** [comment]
@just-boris updated PR with your fixes and suggestions.

**Comment by just-boris** [comment]
:+1: thanks.

I'll keep this PR open, but don't merge it, because my branch `new-face` is going to die

**Comment by baev** [comment]
@sskorol new face was merged into master, you can rebase and send pull request to https://github.com/allure-framework/allure-core

**Comment by sskorol** [comment]
@just-boris @baev created [#777](https://github.com/allure-framework/allure-core/pull/777)

**Pull request closed without merging**